src/services/music-providers/youtube-music/youtube-music.service.ts

```diff
@@ -0,0 +1,286 @@
+        let result: string[] = [];
+
+        for (const artist of artists) {
+            if (artist.id) {
+                result.push(artist.name);
+                continue;
+            }
+
+            const names = artist.name.replace(/, & | & | and | и /g, ', ');
+            result = [...result, ...names.split(', ')];
+        }
+
+        return result;
+    }
+
+    private checkThatTrackIsSuitable(
+        findName: string,
+        findArtist: string[],
+        foundName: string,
+        foundArtists: string[],
+    ) {
+        findName = findName.toLowerCase();
+        findArtist = findArtist.map((a) => a.toLowerCase());
+        foundName = foundName.toLowerCase();
+        foundArtists = foundArtists.map((a) => a.toLowerCase());
+
+        const findNameWithoutParticipantDesignations =
+            this.getTrackNameWithoutParticipantDesignations(findName);
+        const foundNameWithoutParticipantDesignations =
+            this.getTrackNameWithoutParticipantDesignations(foundName);
+
+        if (
+            findNameWithoutParticipantDesignations !==
+            foundNameWithoutParticipantDesignations
+        ) {
+            return false;
+        }
+
+        const featuringArtists = findArtist.filter(
+            (a) => foundName.indexOf(a) !== -1,
+        );
+
+        const allFoundArtists = [
+            ...new Set([...foundArtists, ...featuringArtists]),
+        ];
+        return checkIfArraysAreEqual<string>(findArtist, allFoundArtists);
+    }
+
+    private getTrackNameWithoutParticipantDesignations(name: string): string {
+        const regex = /\s+\((ft|feat|prod)\./;
+
+        return name.split(regex)[0].trim();
+    }
+
+    private async getPlaylist(
+        playlistId: string,
+        options?: GetPlaylistOptions,
+    ): Promise<PlaylistLibmuse> {
+        return retry<PlaylistLibmuse>(
+            async () => await get_playlist(playlistId, options),
+            async () =>
+                this.notifyThatFunctionCompletedWithError('getPlaylist'),
+        );
+    }
+
+    private async search(
+        query: string,
+        options?: SearchOptions,
+    ): Promise<SearchResults> {
+        return retry<SearchResults>(
+            async () => await search(query, options),
+            async () => this.notifyThatFunctionCompletedWithError('search'),
+        );
+    }
+
+    private async removePlaylistItemsWrapperOverLibmuseMethod(
+        playlistId: string,
+        videoIds: {
+            videoId: string;
+            setVideoId: string;
+        }[],
+    ): Promise<EditPlaylistResult> {
+        return retry<EditPlaylistResult>(
+            async () => await remove_playlist_items(playlistId, videoIds),
+            async () =>
+                this.notifyThatFunctionCompletedWithError(
+                    'removePlaylistItemsWrapperOverLibmuseMethod',
+                ),
+        );
+    }
+
+    private notifyThatFunctionCompletedWithError(functionName: string) {
```

Comment:
Consider renaming the method `notifyThatFunctionCompletedWithError` to something more concise and meaningful, such as `logErrorAndRetry`. This will improve readability and better reflect the method's purpose: logging an error and initiating a retry.